src/components/UserActions/UserActions.ts

```diff
@@ -231,6 +261,31 @@ export class UserActions extends Component {
         this.element.innerHTML = '';
         this.element.appendChild(element);
     }
+
+    private showViewedByCustomer() {
+        this.bind.onRootElement(ResultListEvents.newResultDisplayed, (args: IDisplayedNewResultEventArgs) => {
+            if (Boolean(args.item.getElementsByClassName('CoveoViewedByCustomer').length)) {
+                return;
+            }
+            const viewedByCustomerElement = document.createElement('span');
+            new ViewedByCustomer(viewedByCustomerElement, undefined, this.bindings, args.result);
+            args.item.querySelector('.coveo-result-row:last-child .coveo-result-cell').appendChild(viewedByCustomerElement);
+        });
+    }
+
+    private tagViewsOfUser() {
+        Coveo.$$(this.root).on('buildingQuery', (e, args) => {
+            try {
+                const createdBy = this.options.record.fields.CreatedBy.value.fields.Email.value;
```

Comment:
Hm no-no. This is some Salesforce logic, we don't want that into the JSUI-Extension
Instead of passing the record, only get the 'createdBy' instead